tests/test_auto3d_data_analyzer.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import unittest
+from os import makedirs, path, remove, rmdir
+
+import nibabel as nib
+import numpy as np
+import torch
+
+from monai.apps.auto3d.data_analyzer import DataAnalyzer
+from monai.data import create_test_image_3d
+
+device = "cuda" if torch.cuda.is_available() else "cpu"
+n_workers = 0 if sys.platform in ("win32", "darwin") else 2
+
+
+class TestDataAnalyzer(unittest.TestCase):
+    def test_data_analyzer(self):
+        source_datalist = {
+            "testing": [{"image": "val_001.fake.nii.gz"}],
+            "training": [
+                {"fold": 0, "image": "tr_image_001.fake.nii.gz", "label": "tr_label_001.fake.nii.gz"},
+                {"fold": 0, "image": "tr_image_002.fake.nii.gz", "label": "tr_label_002.fake.nii.gz"},
+                {"fold": 1, "image": "tr_image_001.fake.nii.gz", "label": "tr_label_001.fake.nii.gz"},
+                {"fold": 1, "image": "tr_image_004.fake.nii.gz", "label": "tr_label_004.fake.nii.gz"},
+            ],
+        }
+
+        # Generate datasets
+        tmp_dir = "tests/testing_data/auto3d_tmp"
```

Comment:
I would suggest to use `with tempfile.TemporaryDirectory() as tempdir` instead, refer to:
https://github.com/Project-MONAI/MONAI/blob/dev/tests/test_bundle_init_bundle.py#L26

Thanks.

Reply:
Now commit 56bbc32afe4eec4e7763129d0513f9890a38eb7f uses the tempfile package